### PR TITLE
server: fix err check

### PR DESCRIPTION
**Pull request by Tjianke**
### What problem does this PR solve?

Issue Number: #22979  

Problem Summary:

### What is changed and how it works?

What's Changed: added err check

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: server/tidb_test.go [hunk 1]
```diff
@@ -485,7 +512,10 @@ func registerTLSConfig(configName string, caCertPath string, clientCertPath stri
 		ServerName:         serverName,
 		InsecureSkipVerify: !verifyServer,
 	}
-	mysql.RegisterTLSConfig(configName, tlsConfig)
+	err = mysql.RegisterTLSConfig(configName, tlsConfig)
+	if err != nil {
+		panic(err)
```
Comment (xhebox) [inline server/tidb_test.go:517]: Should not panic, return it.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Comment by xhebox** [comment]
Ask for reviewer in the slack channel #sig-sql-infra

**Review by xiongjiwei — APPROVED** [review]

File: server/tidb_test.go [hunk 2]
```diff
@@ -485,7 +512,10 @@ func registerTLSConfig(configName string, caCertPath string, clientCertPath stri
 		ServerName:         serverName,
 		InsecureSkipVerify: !verifyServer,
 	}
-	mysql.RegisterTLSConfig(configName, tlsConfig)
+	err = mysql.RegisterTLSConfig(configName, tlsConfig)
+	if err != nil {
+		return err
+	}
```
Comment (xiongjiwei) [inline server/tidb_test.go:518]: a little wordy, `return mysql.RegisterTLSConfig(configName, tlsConfig)` better?

Comment (Tjianke) [reply server/tidb_test.go:518]: Done.

File: server/tidb_test.go [hunk 3]
```diff
@@ -485,7 +512,10 @@ func registerTLSConfig(configName string, caCertPath string, clientCertPath stri
 		ServerName:         serverName,
 		InsecureSkipVerify: !verifyServer,
 	}
-	mysql.RegisterTLSConfig(configName, tlsConfig)
+	err = mysql.RegisterTLSConfig(configName, tlsConfig)
+	if err != nil {
+		return mysql.RegisterTLSConfig(configName, tlsConfig)
+	}
 	return nil
```
Comment (xiongjiwei) [inline server/tidb_test.go:519]: sorry, I do not make it clear.
```suggestion
	return mysql.RegisterTLSConfig(configName, tlsConfig)
```

**Pull request merged**